src/marqo/s2_inference/hf_utils.py

```diff
@@ -229,16 +233,25 @@ def extract_huggingface_archive(path: str) -> str:
             # return the path to the new directory
             return new_dir
         except (tarfile.ReadError, zipfile.BadZipfile):
-            raise InvalidModelPropertiesError(f'Marqo encountered an error while extracting the compressed model archive from {path}.\n '
-                                              f'This is probably because the file is corrupted or the extionsion {ext} is not supported. '
+            try:
+                os.remove(path)
+            except Exception as remove_e:
```

Comment:
Let's stop excepting all Exceptions. We need to make this more specific. Excepting exceptions can prevent the proper error from surfacing. This changes some errors that should be internal (should be 5xx errors) into user errors (4xx errors). Check the Exception's inheritance to find the status code: https://github.com/marqo-ai/marqo/blob/188fe384c3228c02eb3a8a37bf0788e1c3345ff6/src/marqo/errors.py#LL60C7-L60C20
 
Errors that aren't caught by the API's [exception handler](https://github.com/marqo-ai/marqo/blob/188fe384c3228c02eb3a8a37bf0788e1c3345ff6/src/marqo/tensor_search/api.py#L68), such as RunTimeErrors, appear as 500 errors